graphics/svg_surface.cpp

```diff
@@ -18,76 +18,50 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-#include "common/scummsys.h"
+#include "graphics/svg_surface.h"
+
+#include "common/endian.h"
 #include "common/stream.h"
 #include "common/textconsole.h"
-
-#include "graphics/managed_surface.h"
 #include "graphics/pixelformat.h"
-
-#include "graphics/svg.h"
-
-
 #define NANOSVG_IMPLEMENTATION
 #include "graphics/nanosvg/nanosvg.h"
 #define NANOSVGRAST_IMPLEMENTATION
 #include "graphics/nanosvg/nanosvgrast.h"
 
-namespace Graphics {
-
-SVGBitmap::SVGBitmap(Common::SeekableReadStream *in) {
-	int32 size = in->size();
-	char *data = (char *)malloc(size + 1);
-
-	in->read(data, size);
-	data[size] = '\0';
-
-	_svg = nsvgParse(data, "px", 96);
-	free(data);
-
-	if (_svg == NULL)
-		error("Cannot parse SVG image");
-
-	_rasterizer = NULL;
-	_render = NULL;
-
 #ifdef SCUMM_BIG_ENDIAN
-	_pixelformat = new Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0);
+const Graphics::PixelFormat PIXELFORMAT = Graphics::PixelFormat(4, 8, 8, 8, 8, 24, 16, 8, 0);
```

Comment:
Right. I've changed it in a slightly different way, using a `#define` as it serves basically the same purpose.

Otherwise I can inline the `#ifdef` in the constructor but it wont look as good.